runtime/src/main/java/org/corfudb/runtime/view/workflows/AddNode.java

```diff
@@ -46,7 +47,7 @@ public AddNode(@NonNull String endpointToAdd, @NonNull CorfuRuntime runtime,
     protected UUID sendRequest(@NonNull ManagementClient managementClient) throws TimeoutException {
         // Bootstrap a management server first.
         Layout layout = new Layout(runtime.getLayoutView().getLayout());
-        runtime.getManagementView().bootstrapManagementServer(nodeForWorkflow, layout).join();
+        CFUtils.getUninterruptibly(runtime.getManagementView().bootstrapManagementServer(nodeForWorkflow, layout));
```

Comment:
You can pass Network/Timeout exceptions types.

Reply:
Ok, I see. I'll pass the Network and TimeOut types here and return the 3 exceptions.